File: src/main/java/io/github/toberocat/guiengine/view/GuiView.java
package io.github.toberocat.guiengine.view;

import io.github.toberocat.guiengine.context.GuiContext;
import org.bukkit.inventory.Inventory;
import org.jetbrains.annotations.NotNull;

/**
 * A record class representing a GUI view in a Minecraft plugin.
 */
public record GuiView(@NotNull Inventory inventory, @NotNull GuiContext context) {
}

File: src/main/kotlin/io/github/toberocat/guiengine/view/GuiView.kt
package io.github.toberocat.guiengine.view

import io.github.toberocat.guiengine.context.GuiContext
import org.bukkit.inventory.Inventory

/**
 * A record class representing a GUI view in a Minecraft plugin.
 */
data class GuiView(val inventory: Inventory, val context: GuiContext)
